src/mlpack/core/optimizers/cmaes/cmaes_impl.hpp

```diff
@@ -0,0 +1,765 @@
+      if (diag)
+        sum = tempRandom[i];
+      else
+      {
+        sum = double(0);
+        for (int j = 0; j < N; ++j)
+          sum += B(i,j)*tempRandom[j];
+      }
+      ps[i] = invps*ps[i] + sqrtFactor*sum;
+    }
+
+    // calculate norm(ps)^2
+    double psxps = std::pow(arma::norm(ps),2);
+
+    // cumulation for covariance matrix (pc) using B*D*z~N(0,C)
+    int hsig = std::sqrt(psxps) / std::sqrt(double(1) - std::pow(double(1) - cs, double(2)* gen))
+        / chiN < double(1.4) + double(2) / (N + 1);
+    const double ccumcovinv = 1.-ccumcov;
+    const double hsigFactor = hsig*std::sqrt(ccumcov*(double(2)-ccumcov));
+    for (int i = 0; i < N; ++i)
+      pc[i] = ccumcovinv*pc[i] + hsigFactor*BDz[i];
+
+    // update of C
+    adaptC2(hsig);
+
+    // update of sigma
+    sigma *= std::exp(((std::sqrt(psxps) / chiN) - double(1))* cs / damps);
+
+    state = UPDATED;
+  }
+
+  /**
+   * Some stopping criteria can be set in initials.par, with names starting
+   * with stop... Internal stopping criteria include a maximal condition number
+   * of about 10^15 for the covariance matrix and situations where the numerical
+   * discretisation error in x-space becomes noticeably. You can get a message
+   * that contains the matched stop criteria via getStopMessage().
+   * @return Does any stop criterion match?
+   */
+   template<typename funcType>
+   bool CMAES<funcType>::testForTermination()
+  {
+    double range, fac;
+    int iAchse, iKoo;
+    int diag = diagonalCov == 1 || diagonalCov >= gen;
+
+    bool end = false;
+
+    // function value reached
+    if ((gen > 1 || state > SAMPLED) && stStopFitness.flg &&
+        functionValues[(int)index[0]] <= stStopFitness.val)
+    {
+      Log::Info << "Fitness: function value " << functionValues[(int)index[0]]
+          << " <= stopFitness (" << stStopFitness.val << ")" << std::endl;
+      end = true;
+    }
+
+    // TolFun
+    range = std::max(maxElement(funcValueHistory, (int) std::min(gen, (double)funcValueHistory.size())),
+        arma::max(functionValues)) -
+        std::min(minElement(funcValueHistory, (int) std::min(gen, (double)funcValueHistory.size())),
+        arma::min(functionValues));
+
+    if (gen > 0 && range <= stopTolFun)
+    {
+       Log::Info << "TolFun: function value differences " << range
+          << " < stopTolFun=" << stopTolFun << std::endl;
+       end = true;
+    }
+
+    // TolFunHist
+    if (gen > funcValueHistory.size())
+    {
+      range = maxElement(funcValueHistory, (int)funcValueHistory.size())
+          - minElement(funcValueHistory, (int)funcValueHistory.size());
+      if (range <= stopTolFunHist)
+         Log::Info << "TolFunHist: history of function value changes " << range
+            << " stopTolFunHist=" << stopTolFunHist << std::endl;
+         end = true;
+    }
+
+    // TolX
+    int cTemp = 0;
+    for (int i = 0; i < N; ++i)
+    {
+      cTemp += (sigma*std::sqrt(C(i,i)) < stopTolX) ? 1 : 0;
```

Comment:
arma::uvec x = arma::find((sigma*arma::sqrt(C.diag())) < stopTolX);
arma::uvec y = arma::find(sigma*pc < stopTolX);
int cTemp = x.n_rows + y.n_rows;

replaced all by this .. i think @zoq there can be a better implementation of this which im not aware of ..
Done